test/test_transforms_v2.py

```diff
@@ -1283,13 +1296,24 @@ def test_transform(self, make_input, device):
     @pytest.mark.parametrize(
         "fn", [F.horizontal_flip, transform_cls_to_functional(transforms.RandomHorizontalFlip, p=1)]
     )
-    def test_image_correctness(self, fn):
-        image = make_image(dtype=torch.uint8, device="cpu")
-
+    @pytest.mark.parametrize(
+        "make_input",
+        [
+            make_image,
+            pytest.param(
+                make_image_cvcuda,
+                marks=pytest.mark.skipif(not CVCUDA_AVAILABLE, reason="CVCUDA is not available"),
+            ),
+        ],
+    )
+    def test_image_correctness(self, fn, make_input):
+        image = make_input()
         actual = fn(image)
-        expected = F.to_image(F.horizontal_flip(F.to_pil_image(image)))
-
-        torch.testing.assert_close(actual, expected)
+        if make_input is make_image_cvcuda:
+            image = F.cvcuda_to_tensor(image)[0]  # Remove batch dimension: [1, C, H, W] -> [C, H, W]
+        expected = F.horizontal_flip(F.to_pil_image(image))
+        # CV-CUDA tensors are on CUDA, PIL images are on CPU, so disable device checking
+        assert_equal(actual, expected, check_device=False)
```

Comment:
Let's not do that, it weakens the check. It's safer and more explicit to just call `.cpu()` on `image` above:

```py
image = F.cvcuda_to_tensor(image)[0].cpu()
```